boofuzz/primitives/numbers.py

```diff
@@ -0,0 +1,150 @@
+        @type  fuzzable:      bool
+        @param fuzzable:      (Optional, def=True) Enable/disable fuzzing of this primitive
+        @type  name:          str
+        @param name:          (Optional, def=None) Specifying a name gives you direct access to a primitive
+        """
+
+        super(Numbers, self).__init__()
+
+        self._value = self._original_value = value
+        self.size = size
+        self.max_num = max_num
+        self.signed = signed
+        self._fuzzable = fuzzable
+        self._name = name
+
+        if not self._fuzz_library:
+            self._fuzz_library = [
+                # taken from https://github.com/minimaxir/big-list-of-naughty-strings
+                "0",
+                "1",
+                "1.00",
+                "$1.00",
+                "1/2",
+                "1E2",
+                "1E02",
+                "1E+02",
+                "1/0",
+                "0/0",
+                "0.00",
+                "0..0",
+                ".",
+                "0.0.0",
+                "0,00",
+                "0,,0",
+                ",",
+                "0,0,0",
+                "0.0/0",
+                "1.0/0.0",
+                "0.0/0.0",
+                "1,0/0,0",
+                "0,0/0,0",
+                "999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999",
+                "NaN",
+                "Infinity",
+                "INF",
+                "1#INF",
+                "1#QNAN",
+                "1#SNAN",
+                "1#IND",
+                "0x0",
+                "0xffffffff",
+                "0xffffffffffffffff",
+                "0xabad1dea",
+                "123456789012345678901234567890123456789",
+                "1,000.00",
+                "1 000.00",
+                "1'000.00",
+                "1,000,000.00",
+                "1 000 000.00",
+                "1'000'000.00",
+                "1.000,00",
+                "1 000,00",
+                "1'000,00",
+                "1.000.000,00",
+                "1 000 000,00",
+                "1'000'000,00",
+                "01000",
+                "08",
+                "09",
+                "2.2250738585072011e-308",
+            ]
+            signed_nums = [
+                "-1",
+                "-1.00",
+                "-$1.00",
+                "-1/2",
+                "-1E2",
+                "-1E02",
+                "-1E+02",
+                "-2147483648/-1",
+                "-9223372036854775808/-1",
+                "-0",
+                "-0.0",
+                "+0",
+                "+0.0",
+                "--1",
+                "-",
+                "-.",
+                "-,",
+                "-Infinity",
+                "-1#IND",
+            ]
+
+            if signed:
+                for num in signed_nums:
+                    self._fuzz_library.append(num)
+
+            bits = 64
```

Comment:
Fixed field size conflicts with
https://github.com/jtpereyda/boofuzz/blob/650d41300375dd05e7894175ffe377ba613abc3d/boofuzz/primitives/numbers.py#L22

Too long values should either be truncated later on or not be generated in the first place.

Reply:
Yeah, it was some quick code that I didn't want to think much of, but I guess I could use something like `bits = 64 if size == -1 else size`.